crates/bevy_ui/src/widget/image.rs

```diff
@@ -1,41 +1,48 @@
 use crate::{
-    measurement::AvailableSpace, ContentSize, Measure, Node, UiImage, UiTextureAtlasImage,
+    measurement::AvailableSpace, ContentSize, Measure, Node, UiImage, UiScale, UiTextureAtlasImage,
 };
 use bevy_asset::{Assets, Handle};
+
 #[cfg(feature = "bevy_text")]
 use bevy_ecs::query::Without;
 use bevy_ecs::{
     prelude::Component,
     query::With,
     reflect::ReflectComponent,
-    system::{Query, Res},
+    system::{Local, Query, Res},
 };
 use bevy_math::Vec2;
 use bevy_reflect::{std_traits::ReflectDefault, FromReflect, Reflect, ReflectFromReflect};
 use bevy_render::texture::Image;
 use bevy_sprite::TextureAtlas;
 #[cfg(feature = "bevy_text")]
 use bevy_text::Text;
+use bevy_window::{PrimaryWindow, Window};
 
-/// The size of the image in physical pixels
+/// The size of the image's texture
 ///
-/// This field is set automatically by `update_image_calculated_size_system`
+/// This component is updated automatically by `update_image_calculated_size_system`
 #[derive(Component, Debug, Copy, Clone, Default, Reflect, FromReflect)]
 #[reflect(Component, Default, FromReflect)]
 pub struct UiImageSize {
+    /// The size of the image's texture
+    ///
+    /// This field is updated automatically by `update_image_calculated_size_system`
     size: Vec2,
 }
 
 impl UiImageSize {
+    /// The size of the image's texture
     pub fn size(&self) -> Vec2 {
         self.size
     }
 }
 
 #[derive(Clone)]
+/// Used to calculate the size of UI image nodes
 pub struct ImageMeasure {
-    // target size of the image
-    size: Vec2,
+    /// The size of the image's texture
+    pub size: Vec2,
```

Comment:
Do we want this to be editable by the user? Or should this be `pub(crate)`?

Reply:
Yes it's intended to be `pub`. That way someone can use `ImageMeasure` for their own custom widget if they want.

Otherwise, it doesn't matter that it's `pub` as MeasureFuncs can't be accessed by users once they've been added to the layout tree.